### PR TITLE
Optimize References Loaders Display

### PR DESCRIPTION
**Pull request by djhi**
When displaying a lot of references fields/inputs, for example in a List, having a lot of loaders can harm performances, especially on low-end devices. This PR applies the same optimization as for the skeletons: don't display loaders until at least a second has passed

### CODE REVIEW

File: packages/ra-ui-materialui/src/layout/LinearProgressIfTooLong.tsx [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { useTimeout } from 'ra-core';
+import * as React from 'react';
+import LinearProgress from './LinearProgress';
+
+export const LinearProgressIfTooLong = ({ timeout = 1000, ...props }) => {
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/layout/LinearProgressIfTooLong.tsx:5]: I think we don't need a second component for that. It should be the default behavior of our LinerProgress

**Review by fzaninotto — CHANGES_REQUESTED** [review]

File: packages/ra-ui-materialui/src/field/ReferenceArrayField.tsx [hunk 2]
```diff
@@ -16,6 +15,7 @@ import {
 import { fieldPropTypes, PublicFieldProps, InjectedFieldProps } from './types';
 import { ClassesOverride } from '../types';
 import sanitizeFieldRestProps from './sanitizeFieldRestProps';
+import { LinearProgress } from '../layout';
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/field/ReferenceArrayField.tsx:18]: useless change

File: packages/ra-ui-materialui/src/field/ReferenceField.tsx [hunk 3]
```diff
@@ -15,11 +15,11 @@ import {
     Record,
 } from 'ra-core';
 
-import LinearProgress from '../layout/LinearProgress';
 import Link from '../Link';
 import sanitizeFieldRestProps from './sanitizeFieldRestProps';
 import { PublicFieldProps, fieldPropTypes, InjectedFieldProps } from './types';
 import { ClassesOverride } from '../types';
+import { LinearProgress } from '../layout';
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/field/ReferenceField.tsx:22]: useless change

File: packages/ra-ui-materialui/src/input/ReferenceArrayInput.tsx [hunk 4]
```diff
@@ -14,7 +14,7 @@ import {
 } from 'ra-core';
 
 import sanitizeInputRestProps from './sanitizeInputRestProps';
-import LinearProgress from '../layout/LinearProgress';
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/input/ReferenceArrayInput.tsx:17]: useless change

File: packages/ra-ui-materialui/src/input/ReferenceInput.tsx [hunk 5]
```diff
@@ -17,9 +17,9 @@ import {
 } from 'ra-core';
 
 import sanitizeInputRestProps from './sanitizeInputRestProps';
-import LinearProgress from '../layout/LinearProgress';
 import Labeled from './Labeled';
 import ReferenceError from './ReferenceError';
+import { LinearProgress } from '../layout';
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/input/ReferenceInput.tsx:22]: useless change

**Review by fzaninotto — CHANGES_REQUESTED** [review]

File: packages/ra-ui-materialui/src/field/ReferenceArrayField.tsx (hunk 2 above)
Comment (djhi) [reply packages/ra-ui-materialui/src/field/ReferenceArrayField.tsx:18]: Nope, we are now using our own LinearProgress instead of MUI

**Review by fzaninotto — APPROVED** [review]

**Pull request merged**